client/app/lib/components/table/TanStackTableBuilder/useTanStackTableBuilder.tsx

```diff
@@ -70,6 +71,9 @@ const useTanStackTableBuilder = <D extends object>(
       columnFilters,
       globalFilter: searchKeyword.trim(),
       pagination,
+      columnVisibility: Object.fromEntries(
+        props.columns.map((column) => [column.title, column.visible ?? true]),
```

Comment:
All of the other props like sortable, filterable, searchable are false by default.  
You should consider making the prop called `hidden` (false by default) just to align with existing behaviour.